tests/test_rule_based.py

```diff
@@ -0,0 +1,135 @@
+import pytest
+from tocount.rule_based import universal_tokens_estimator, openai_tokens_estimator_gpt_3_5, openai_tokens_estimator_gpt_4
+
+
+def test_universal_text_with_simple_prompt():
```

Comment:
I recommend directly testing the `estimate_text_tokens` function rather than the inner functions.